pkg/devfile/convert/convert.go

```diff
@@ -278,6 +330,57 @@ func setDevfileComponentsForS2I(d data.DevfileData, s2iImage string, localConfig
 		volumeMounts = append(volumeMounts, volumeMount)
 	}
 
+	// Add volumes for:
+	// - /opt/app-root
+	// - ${ODO_S2I_DEPLOYMENT_DIR} if outside of /opt/app-root
+
+	var deploymentDir string
+	for _, env := range s2iEnv {
+		if env.Name == "ODO_S2I_DEPLOYMENT_DIR" {
+			deploymentDir = env.Value
+			break
+		}
+	}
+
+	separateDeploymentsMount := len(deploymentDir) > 0 && !strings.HasPrefix(deploymentDir, occlient.DefaultAppRootDir)
```

Comment:
Can you move this code to L356 ? I think it will make it easier to understand the code.